src/probnum/problems/__init__.py

```diff
@@ -1 +1,5 @@
 """Definitions and collection of problems solved by probabilistic numerical methods."""
+
+from ._problems import *
```

Comment:
Star imports are discouraged since they potentially import unwanted modules, functions or classes. Instead explicitly write out the classes and functions you are importing. For example currently `probnum.diffeq.filtsmooth` exports numpy as `probnum.diffeq.filtsmooth.np`:

```python
dir(probnum.diffeq.odefiltsmooth)
Out[4]: 
['GaussianIVPFilter',
 '__builtins__',
 '__cached__',
...
 'ivp2filter',
 'np',
 'pnfs',
 ...
```